### PR TITLE
One big group

### PR DESCRIPTION
**Pull request by JasonHoffmann**
This takes a slightly different approach from #5. Instead of creating new groups for each set of fields, this replaces the existing field with a _single_ group, then swaps fields in and out to render data.

Some advantages there:
- Data in the form is not double-nested
- No need to overwrite meta values since the group is structured properly
- Can rely on CMB2 for more JS functionality, including removing rows and WYSIWYG initialization
- Can pre-render WYSIWYG fields with data and template, so don't need to try to clone settings from existing WYSIWYG fields or output a WYSIWYG template on AJAX requests
- Can use more of CMB2 classes and styles
- Less code

The disadvantage being that the CMB2 Field instance has to be cloned and recreated fairly often, but that is really just building up an array, and it is more or less the same as creating a new group for each individual data block.

### CODE REVIEW

**Comment by JasonHoffmann** [comment]
Also, something that would make this way easier in CMB2. If there was a way to edit a fields `field_options` parameter (it's currently `protected`), I wouldn't need to clone the group so often, since I can just populate the field_options array with the data I need.

**Comment by jtsternberg** [comment]
https://github.com/CMB2/CMB2/commit/1882114a0450d7b655230600ccdb954e72fcb03f should help with that. Will just need to modify the field's `options` param before calling `CMB2_Field::set_options()` again. You can update the `options` param, by doing something like:
```php
$field->set_prop( 'options', $new_modified_options );
$field->set_options();
```

**Comment by JasonHoffmann** [comment]
Yea the problem was trying to modify it after `set_options` was already called. But that gives me an idea about storing some things beforehand...

**Comment by JasonHoffmann** [comment]
Oh and thanks for the commit @jtsternberg! Should make things easier.

**Comment by jtsternberg** [comment]
> Yea the problem was trying to modify it after set_options was already called.

Right, but now you can call set_options at any arbitrary time, to reset the field options array.

**Comment by JasonHoffmann** [comment]
Oh right, perfect!

File: cmb2-flexible-content-field.php [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+				'id' => $field_id,
+				'type' => 'group',
+				'context' => 'normal',
+				'repeatable' => true,
+				'fields' => array(),
+			);
+			$field_group = $flexible_field->get_field_clone( $group_args );
+			$field_group->data_to_save = $values;
+
+			// // The saved array is used to hold sanitized values.
```
Comment (jjeaton) [inline cmb2-flexible-content-field.php]: Extra comment marks

**Pull request merged**